ArjunaJTS/jtax/classes/com/arjuna/ats/internal/jta/transaction/jts/jca/XATerminatorImple.java

```diff
@@ -452,7 +452,13 @@ public boolean beforeCompletion (Xid xid) throws javax.transaction.SystemExcepti
 
     public Transaction getTransaction(Xid xid) throws XAException {
         // first see if the xid is a root coordinator
-        return com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple.getTransaction(new XidImple(xid).getTransactionUid());
+        Transaction transaction = com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple
```

Comment:
I agree, this was a cut and paste error. The original implementation was for JTA and the JTS version was only added because it was in the interface. It needs changing to 
```java
com.arjuna.ats.internal.jta.transaction.jts.subordinate.jca.TransactionImple
```
Just change *all* usages (in this file) of com.arjuna.ats.internal.jta.transaction.arjunacore.TransactionImple to TransactionImple (the one in the java import statement).